components/prism-stylus.js

```diff
@@ -0,0 +1,61 @@
+Prism.languages.stylus = {
+	'comment': {
+		pattern: /(^|[^\\])(\/\*[\w\W]*?\*\/|\/\/.*?(\r?\n|$))/g,
+		lookbehind: true
+	},
+	'keyword': /(px|r?em)\b|\b(is|defined|not|isnt|and|or|unless|for|in)\b/g,
+	'atrule': /@[\w-]+(?=\s+\S+)/gi,
+	'url': /url\((["']?).*?\1\)/gi,
+	'variable': /^\s*([\w-]+)(?=\s*[+-\\]?=)/gm,
+	'string': /("|')(\\\n|\\?.)*?\1/g,
+	'important': /\B!important\b/gi,
+	'hexcode': /#[\da-f]{3,6}/gi,
+	'entity': /\\[\da-f]{1,8}/gi,
+	'number': /\d+\.?\d*%?/g,
+	'selector': [
+		{
+			pattern: /:{1,2}after|before|first-letter|first-line|selection/g,
+			alias: 'pseudo-element'
+		},{
+			pattern: /:(?:active|checked|disabled|empty|enabled|first-child|first-of-type|focus|hover|in-range|invalid|lang|last-child|last-of-type|link|not|nth-child|nth-last-child|nth-last-of-type|nth-of-type|only-of-type|only-child|optional|out-of-range|read-only|read-write|required|root|target|valid|visited)(?:\(.*\))?/g,
+			alias:'pseudo-class'
```

Comment:
Is it important to distinguish between pseudo-elements and pseudo-classes?
It would probably be more future-proof to match them both with a more generic pattern like `/::?[\w-]+(?:\(.*\))?/` or something.

I'd like to hear @LeaVerou's opinion on this.